packages/presentation/lib/screen/main_my_bloc.dart

```diff
@@ -55,35 +56,44 @@ class MainBlocImpl extends BlocImpl implements MainBloc {
   }
 
   @override
-  void generateRandomNumber() {
-    _screenData.randomNumber = _generateNumberUseCase();
-    _screenData.isDisabled = false;
-    _updateData();
+  void startNewGame() {
+    _screenData = _screenData.copyWith(
+      randomNumber: _generateNumberUseCase(),
+      counter: 0,
+      isDisabled: false,
+    );
+    _updateData(data: _screenData);
   }
 
+  @override
   void checkNumber() {
-    number = int.tryParse(_editController.text)!;
+    number = int.tryParse(_editController.text) ?? 0;
     final params = ComparedNumbers(
       guessNumber: number,
       randomNumber: _screenData.randomNumber,
       counter: _screenData.counter,
     );
     final currentState = _checkStateUseCase(params);
     if (currentState is OutOfAttempts) {
-      _showAlert(
-          'You Out Of Attempts \nPlease, start new game.');
-      _screenData.counter = 1;
-      _screenData.isDisabled = true;
-      _updateData();
+      _showAlert('You Out Of Attempts \nPlease, start new game.');
+      _screenData = _screenData.copyWith(
+        counter: 0,
+        isDisabled: true,
+      );
+      _updateData(data: _screenData);
     } else if (currentState is YouWon) {
       _showAlert('You Won');
-      _screenData.counter = 1;
-      _screenData.isDisabled = true;
-      _updateData();
+      _screenData = _screenData.copyWith(
+        counter: 0,
+        isDisabled: true,
+      );
+      _updateData(data: _screenData);
     } else {
+      _screenData = _screenData.copyWith(
+        counter: _screenData.counter + 1,
+      );
```

Comment:
why do u need copyWith?
Seems prettier:
screenData.counter = _screenData.counter + 1;
then call _updateData()

Reply:
To create copy of existing object, not to change it. But seems at the end at = we finally changing it? @sergei-mikhailovskii-idf idea

Reply:
@andrei-lastochkin-idf , Katya is right. state properties should be immutable, while state by itself is mutable

Reply:
@sergei-mikhailovskii-idf ok, seems like it's our local practice on my project